mk/top_of_makefile.mk

```diff
@@ -1,245 +0,0 @@
-ifeq ($(findstring clang,$(CC)),clang)
-DEFAULT_TARGET_ARCH = -target "$(TARGET)"
-endif
-endif
-
-ifeq ($(TARGET_ARCH_NORMAL),x86)
-MARCH = pentium
-MINSTR = 32
-else ifeq ($(TARGET_ARCH_NORMAL),x86_64)
-MARCH = x86-64
-MINSTR = 64
-else
-# TODO: Pass -march and related options. For now, use the default of the
-# toolchain's C compiler.
-# MARCH = $(subst _,-,$(TARGET_ARCH_BASE))
-endif
-
-ifeq ($(TARGET_ABI),eabi)
-MABI = aapcs
-endif
-
-# Cortex-M0, Cortex-M0+, Cortex-M1: armv6_m
-# Cortex-M3: armv7_m
-# Cortex-M4, Cortex-M7: armv7e_m
-ifeq ($(filter-out armv6_m armv7_m armv7e_m,$(TARGET_ARCH_BASE)),)
-MINSTR = thumb
-endif
-
-# Although it isn't mentioned in the GNU Make manual, GNU Make passes
-# $(TARGET_ARCH) in its implicit rules.
-TARGET_ARCH += $(if $(MCPU),-mcpu=$(MCPU)) \
-               $(if $(MARCH),-march=$(MARCH)) \
-               $(if $(MABI),-mabi=$(MABI)) \
-               $(if $(MINSTR),-m$(MINSTR)) \
-               $(NULL)
-
-ifeq ($(CC),)
-$(error You must specify CC)
-endif
-ifeq ($(CXX),)
-$(error You must specify CXX)
-endif
-
-# e.g. "clang-3.6"
-COMPILER_NAME ?= $(notdir $(CC))
-
-# Generate output to a directory like build/x86_64-unknown-linux-elf-clang-3.6.
-BUILD_PREFIX_PRIMARY ?= build
-BUILD_PREFIX_SUB ?= $(TARGET)-$(COMPILER_NAME)
-BUILD_PREFIX ?= $(BUILD_PREFIX_PRIMARY)/$(BUILD_PREFIX_SUB)/
-
-EXE_PREFIX ?= $(BUILD_PREFIX)test/ring/
-OBJ_PREFIX ?= $(BUILD_PREFIX)obj/
-LIB_PREFIX ?= $(BUILD_PREFIX)lib/
-
-# GCC 4.6 requires "c1x" and "c++0x" instead of "c11" and "c++11".
-CFLAGS_STD ?= -std=c1x
-CXXFLAGS_STD ?= -std=c++0x
-
-CFLAGS += $(CFLAGS_STD)
-CXXFLAGS += $(CXXFLAGS_STD)
-
-# Add flags needed for ios.
-ifeq ($(TARGET_SYS),ios)
-IOS_MIN_VERSION = 7.0
-ifeq ($(findstring x86,$(TARGET_ARCH_NORMAL)),x86)
-IOS_SDK = iphonesimulator
-IOS_VERSION_FLAG = -mios-simulator-version-min=$(IOS_MIN_VERSION)
-else
-IOS_SDK = iphoneos
-IOS_VERSION_FLAG = -mios-version-min=$(IOS_MIN_VERSION)
-ifeq ($(TARGET_ARCH_BASE),aarch64)
-CFLAGS += -arch arm64
-CXXFLAGS += -arch arm64
-else ifeq ($(TARGET_ARCH_BASE),armv7)
-CFLAGS += -arch $(TARGET_ARCH_BASE) -mfpu=neon
-CXXFLAGS += -arch $(TARGET_ARCH_BASE) -mfpu=neon
-else
-CFLAGS += -arch $(TARGET_ARCH_BASE)
-CXXFLAGS += -arch $(TARGET_ARCH_BASE)
-endif
-endif
-IOS_SDK_PATH = $(shell xcrun --show-sdk-path -sdk $(IOS_SDK))
-CFLAGS += $(DEFAULT_TARGET_ARCH) -isysroot $(IOS_SDK_PATH) $(IOS_VERSION_FLAG)
-CXXFLAGS += $(DEFAULT_TARGET_ARCH) -isysroot $(IOS_SDK_PATH) $(IOS_VERSION_FLAG)
-endif
-
-# Always add full debug info and strip dead code.
-CPPFLAGS += -fpic -fdata-sections -ffunction-sections
-ifeq ($(findstring darwin,$(TARGET_SYS)),darwin)
-# |-gfull| is required for Darwin's |-dead_strip|.
-CPPFLAGS += -gfull
-# libc++ is the default in newer toolchains, but libstdc++ was the default in
-# older toolchains.
-CXXFLAGS += -stdlib=libc++
-LDFLAGS += -fPIC -Wl,-dead_strip
-else
-CPPFLAGS += -g3
-LDFLAGS += -Wl,--gc-sections
-endif
-
-ASFLAGS += -Wa,--noexecstack,-gdwarf
```

Comment:
ASFLAGS is not copied because it seems to be not used (As far as I can tell from the output of a Makefile based build)